packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -142,12 +174,42 @@ class AnalyzeCommand extends FlutterCommand {
       return false;
     }
 
+    // Don't run pub if asking for android analysis.
+    if (boolArg('android')) {
+      return false;
+    }
+
     return super.shouldRunPub;
   }
 
   @override
   Future<FlutterCommandResult> runCommand() async {
-    if (boolArg('suggestions')) {
+    if (boolArg('android')) {
+      final AndroidAnalyzeOption option;
+      if (argResults!['list-build-variants'] as bool) {
+        option = AndroidAnalyzeOption.listBuildVariant;
+      } else if (argResults!['output-app-link-settings'] as bool) {
+        option = AndroidAnalyzeOption.outputAppLinkSettings;
+      } else {
+        throwToolExit('No argument provided to analyze');
```

Comment:
Can you be more clear here? I checked out your branch, tried `flutter analyze --android`, and was confused by this message, and had to look at the code to see what it meant.

Reply:
Although, I suppose, this isn't intended to be user facing. Still, maybe we might regress this in a client that calls this?